engine/jobmaster/dm/api.go

```diff
@@ -98,7 +98,13 @@ func (jm *JobMaster) QueryJobStatus(ctx context.Context, tasks []string) (*JobSt
 					// task finished
 					workerID = workerStatus.ID
 					cfgModRevision = workerStatus.CfgModRevision
-					queryStatusResp = &dmpkg.QueryStatusResponse{Unit: workerStatus.Unit, Stage: metadata.StageFinished}
+					finishedStatus, ok := jm.finishedStatus.Load(taskID)
+					if !ok {
+						queryStatusResp = &dmpkg.QueryStatusResponse{Unit: workerStatus.Unit, Stage: metadata.StageFinished, ErrorMsg: fmt.Sprintf("task %s finished but status not found", taskID)}
```

Comment:
“not found” may cause worry, maybe "task %s is finished and status has been deleted"